diff --git a/src/main.rs b/src/main.rs
index 541de22..db44812 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -71,7 +71,7 @@ fn main() {
         let mut loop_to = 0;
         let mut iterations: usize;
 
-        for index in 0..4 {
+        for index in 0..1 {
             let buffer = Arc::clone(&buffer);
             let back_tree = Arc::clone(&back_tree);
             let buffer_ = buffer.lock().unwrap();
@@ -87,7 +87,9 @@ fn main() {
             std::mem::drop(buffer_);
             loop_to +=  buffer_len / leaf_size as usize / 4 * leaf_size as usize; 
             let handle = thread::spawn(move || {
-            for i in (loop_to-iterations..loop_to+extra).step_by(leaf_size as usize) {
+            //for i in (loop_to-iterations..loop_to+extra).step_by(leaf_size as usize) {
+            for i in (0..buffer_len).step_by(leaf_size as usize) {
+
                 let mut leaf_buf: Vec<u8> = Vec::new();
                 let buffer = buffer.lock().unwrap();
 
